gfx_scene.h

```diff
@@ -1110,473 +1166,267 @@ class GfxSceneInternal
+            it = (metallicity_roughness_map_text != nullptr ? images.find(metallicity_roughness_map_text->basisu_image != nullptr ?
+                metallicity_roughness_map_text->basisu_image : metallicity_roughness_map_text->image) : images.end());
             if(it != images.end())
             {
-                std::map<int32_t, std::pair<GfxConstRef<GfxImage>, GfxConstRef<GfxImage>>>::const_iterator const it2 = maps.find(metallicity_roughness_map);
+                std::map<cgltf_image const*, std::pair<GfxConstRef<GfxImage>, GfxConstRef<GfxImage>>>::const_iterator const it2 =
+                    maps.find(metallicity_roughness_map_text->basisu_image != nullptr ?
+                    metallicity_roughness_map_text->basisu_image : metallicity_roughness_map_text->image);
                 if(it2 != maps.end())
                 {
                     material_ref->roughness_map = (*it2).second.first;
                     material_ref->metallicity_map = (*it2).second.second;
                 }
                 else
                 {
-                    std::string const metallicity_map_file = image_metadata_[(*it).second].asset_file + ".b";
-                    std::string const roughness_map_file = image_metadata_[(*it).second].asset_file + ".g";
+                    size_t const asset_file_ext = image_metadata_[(*it).second].asset_file.rfind('.');
+                    std::string const asset_file_name = image_metadata_[(*it).second].asset_file.substr(0, asset_file_ext);
+                    std::string const asset_file_extension = image_metadata_[(*it).second].asset_file.substr(asset_file_ext);
+                    std::string const metallicity_map_file = asset_file_name + ".metallicity" + asset_file_extension;
+                    std::string const roughness_map_file = asset_file_name + ".roughness" + asset_file_extension;
                     GfxRef<GfxImage> metallicity_map_ref = gfxSceneFindObjectByAssetFile<GfxImage>(scene, metallicity_map_file.c_str());
                     GfxRef<GfxImage> roughness_map_ref = gfxSceneFindObjectByAssetFile<GfxImage>(scene, roughness_map_file.c_str());
-                    if(!metallicity_map_ref || !roughness_map_ref)
+                    if (!metallicity_map_ref)
                     {
+                        if (gfxSceneImport(scene, metallicity_map_file.c_str()) == kGfxResult_NoError)
+                            metallicity_map_ref = gfxSceneFindObjectByAssetFile<GfxImage>(scene, metallicity_map_file.c_str());
+                    }
+                    if (!metallicity_map_ref)
+                    {
+                        if (gfxSceneImport(scene, roughness_map_file.c_str()) == kGfxResult_NoError)
+                            roughness_map_ref = gfxSceneFindObjectByAssetFile<GfxImage>(scene, roughness_map_file.c_str());
+                    }
+                    if(!metallicity_map_ref && !roughness_map_ref)
```

Comment:
This is were we check for standalone already split versions of a image. So if the gltf file specifies a 'xxx.ext' image file it now first checks for a 'xxx.metallicity.ext' and/or 'xxx.roughness.ext' image and loads those directly instead (useful for compressed textures as we cant really separate the channels easily and having them already in separate images allows for each image to use a single channel BC4 compression which results in smaller files/memory and better quality). If no separate images are found then it does the manually splitting as before.

Reply:
Since its possible for a material to only require a single texture for metallicity/roughness (as the other is just a constant) then we only try and split if both images are missing